std/algorithm/iteration.d

```diff
@@ -4280,3 +4284,111 @@ unittest
          [1, 2, 0],
          [2, 1, 0]]));
 }
+
+/**
+Finds the mean (colloquially known as the average) of a range.
+This function has $(D O(r.length)) complexity.
+
+Params:
+    r = an input range
+
+Returns:
+    the mean of r
+*/
+auto mean(R)(R r) if (isInputRange!R && !isInfinite!R)
```

Comment:
That's a lot of typing (and reading). What about just straight-up:
````
is(ElementType!R : double)
````
?

For simplicity, I'd just stick with `double` to minimize precision errors. But if you *really* wanted to, you could parametrize the floating type and use `isFloatingPoint!F` in the sig constraints to enforce it, then the user can specify the type to use for computing the average.